File: python/src/api/handlers/forms/getForm/getFormHandler.py
try:
    import unzip_requirements
except ImportError:
    pass

import boto3
import os
import simplejson as json

dynamodb_client = boto3.resource('dynamodb')
TABLE_NAME = os.environ['FORMS_TABLE_NAME']
DYNAMO_FORMS_ID = os.environ['FORMS_ID']

def create_response(code, message):
  return {
    'statusCode': code,
    'headers': {'Content-Type': 'application/json'},
    'body': json.dumps(message)
  }

def getForm(event, context):

    pk = event['pathParameters']['id']
    print('pk: ', pk)

    pk = DYNAMO_FORMS_ID + '#' + pk
    params = {
        'Key': {
            'PK': pk,
            'SK': pk
        }
    }

    try:
        table = dynamodb_client.Table(TABLE_NAME)
        response = table.get_item(**params)
    except Exception as ex:
        return create_response(500, str(ex))

    print('response: ', response)

    if response['ResponseMetadata']['HTTPStatusCode'] != 200:
      return create_response(500, 'Error getting the Form')

    if not 'Item' in response:
        return create_response(404, 'Form with id: ' + pk + ' not found')

    form = response['Item']

    return create_response(200, form)

File: nodejs/src/api/handlers/forms/getForm/getFormHandler.js
'use strict';

const AWS = require('aws-sdk')
const db = new AWS.DynamoDB.DocumentClient({})

const tableName = process.env.FORMS_TABLE_NAME
const dynamoFormsId = process.env.FORMS_ID

const response = (statusCode, message) => {
    return {
        statusCode: statusCode,
        body: JSON.stringify(message)
    }
}

const getForm = async (event) => {
    const PK = event.pathParameters.id
    console.log('PK: ', PK)
    
    const params = {
      TableName: tableName,
      Key: {
        PK: dynamoFormsId + '#' + PK,
        SK: dynamoFormsId + '#' + PK,
      }
    }
  
    try {
      const formSearch = await db.get(params).promise()
      console.log('formSearch: ', formSearch)
  
      if (!formSearch.Item) {
        return response(404, 'Form with id: ' + PK + 'not found')
      }
  
      const form = formSearch.Item
      return response(200, form)
    } catch (err) {
      console.log('err: ', err)
      return response(500, 'Error getting the Form')
    }
}

module.exports = {
    getForm
}